Authentication/AuthenticationComponentsFactory.swift

```diff
@@ -76,6 +76,10 @@ public struct AuthenticationComponentsFactory: AuthenticationComponentsFactoryTy
         return _loginConfiguration
```

Comment:
In general I think maybe we don't need to have functions that return a factory closure, but directly have the componentfactory's method be the factory closures.
So for example, I would turn:
`func createLoginViewFactory(withDelegate delegate: LoginViewDelegate) -> ([LoginProvider]) -> () -> LoginViewType`
to
`func createLoginViewFactory(withDelegate delegate: LoginViewDelegate, loginProviders: [LoginProvider]) -> LoginViewType`
And if the coordinator needs a closure then we can create one where he calls the ComponentsFactory function.